bokeh/sphinxext/_templates/example_index.rst

```diff
@@ -0,0 +1,8 @@
+:orphan:
+
+.. _example_index:
+
+Example Index
+=============
+
+.. bokeh-example-index::
```

Comment:
I created a template, because I thouht this should be placed at `/gallery/example_index.html`. But because the folder `gallery` will be removed in some building processes, I have to create this somehow. This is not ideal.

Reply:
I think maybe we need to take a step back and disentangle a few things. I don't think we should conflate with curated gallery entries with the blanket generation of pages for every `examples/{plotting, models}` example. I actually think the examples should go under:
```
https://docs.bokeh.org/en/latest/docs/examples/plotting/box_annotation.html
https://docs.bokeh.org/en/latest/docs/examples/models/anscombe.py
```

The indexes for each could be at:
```
https://docs.bokeh.org/en/latest/docs/examples/index.html           # all examples  
https://docs.bokeh.org/en/latest/docs/examples/plotting/index.html  # just plotting examples
https://docs.bokeh.org/en/latest/docs/examples/models/index.html    # just models examples
```

I think you are pretty much already collecting all the info to make this happen. Basically just put the output docs ina  different URL  and also use the metadata to generate those index pages. 

I do understand that *right now* we are "hijacking" the gallery extension to generate all of these standalone pages, and that is OK, for now. But I do think these concerns will eventually need to be separate (another reason I think things will have to move to a custom Sphinx parser for `.py` files at some point) 

Does this make sense and seem feasible? I should add that I don't know what your bandwidth for all this is, if things are getting too involved I can also contribute to this existing PR if need be (just let me know)

Reply:
As a last addition: the motivation for why we should split out examples now is that once we have done that, the gallery can actually be drastically simplified. It can just become references/links to a curated subset of the standalone pages that are now under `/examples`. We can probably get rid of `gallery.json` and maybe even the gallery extensions altogether.